app/src/org/commcare/dalvik/activities/EntityMapActivity.java

```diff
@@ -1,288 +1,186 @@
-import android.location.Address;
-import android.location.Criteria;
-import android.location.Geocoder;
-import android.location.Location;
-import android.location.LocationManager;
 import android.os.Bundle;
 import android.support.v4.content.ContextCompat;
 import android.util.Log;
-
-import com.google.android.maps.GeoPoint;
-import com.google.android.maps.MapActivity;
-import com.google.android.maps.MapView;
-import com.google.android.maps.MyLocationOverlay;
-import com.google.android.maps.OverlayItem;
-
-import org.commcare.android.database.SqlStorage;
-import org.commcare.android.database.user.models.GeocodeCacheModel;
+import android.util.Pair;
+
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.MapFragment;
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.LatLngBounds;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+
+import org.commcare.android.framework.CommCareActivity;
 import org.commcare.android.models.Entity;
 import org.commcare.android.models.NodeEntityFactory;
 import org.commcare.android.util.AndroidInstanceInitializer;
 import org.commcare.android.util.SerializationUtil;
-import org.commcare.dalvik.BuildConfig;
 import org.commcare.dalvik.R;
 import org.commcare.dalvik.application.CommCareApplication;
-import org.commcare.dalvik.geo.EntityOverlay;
-import org.commcare.dalvik.geo.EntityOverlayItemFactory;
 import org.commcare.session.CommCareSession;
 import org.commcare.suite.model.Detail;
-import org.commcare.suite.model.Entry;
 import org.commcare.suite.model.SessionDatum;
 import org.javarosa.core.model.condition.EvaluationContext;
 import org.javarosa.core.model.data.GeoPointData;
 import org.javarosa.core.model.data.UncastData;
 import org.javarosa.core.model.instance.TreeReference;
-import org.javarosa.core.services.storage.StorageFullException;
 
-import java.io.IOException;
-import java.util.List;
-import java.util.NoSuchElementException;
+import java.util.HashMap;
 import java.util.Vector;
 
 /**
  * @author ctsims
  */
-public class EntityMapActivity extends MapActivity {
+@TargetApi(11)
+public class EntityMapActivity extends CommCareActivity implements OnMapReadyCallback,
+        GoogleMap.OnInfoWindowClickListener {
     private static final String TAG = EntityMapActivity.class.getSimpleName();
-    private final static int LOCATION_PERMISSIONS_REQUEST = 1;
+    private static final int MAP_PADDING = 50;  // Number of pixels to pad bounding region of markers
 
-    private MapView map;
-    private MyLocationOverlay mMyLocationOverlay;
-    private Geocoder mGeoCoder;
-    private LocationManager mLocationManager;
-    
+    private EvaluationContext entityEvaluationContext;
     private CommCareSession session;
+    Vector<Entity<TreeReference>> entities;

```

Comment:
this can be made a local variable inside of `onCreate` instead of a class level field